src/event-engine/states/handshake_reconnecting.ts

```diff
@@ -19,34 +33,60 @@ export const HandshakeReconnectingState = new State<HandshakeReconnectingStateCo
 );
 
 HandshakeReconnectingState.onEnter((context) => handshakeReconnect(context));
-HandshakeReconnectingState.onExit(() => reconnect.cancel);
+HandshakeReconnectingState.onExit(() => handshakeReconnect.cancel);
 
-HandshakeReconnectingState.on(reconnectingSuccess.type, (context, event) =>
-  ReceivingState.with(
+HandshakeReconnectingState.on(handshakeReconnectSuccess.type, (context, event) => {
+  const cursor = {
+    timetoken: !!context.cursor?.timetoken ? context.cursor?.timetoken : event.payload.cursor.timetoken,
+    region: event.payload.cursor.region,
+  };
+  return ReceivingState.with(
     {
       channels: context.channels,
       groups: context.groups,
-      cursor: event.payload.cursor,
+      cursor: cursor,
     },
-    [emitEvents(event.payload.events)],
-  ),
-);
+    [emitStatus({ category: categoryConstants.PNConnectedCategory })],
+  );
+});
 
-HandshakeReconnectingState.on(reconnectingFailure.type, (context, event) =>
+HandshakeReconnectingState.on(handshakeReconnectFailure.type, (context, event) =>
   HandshakeReconnectingState.with({ ...context, attempts: context.attempts + 1, reason: event.payload }),
 );
 
-HandshakeReconnectingState.on(reconnectingGiveup.type, (context) =>
-  HandshakeFailureState.with({
-    groups: context.groups,
-    channels: context.channels,
-    reason: context.reason,
-  }),
+HandshakeReconnectingState.on(handshakeReconnectGiveup.type, (context, event) =>
+  HandshakeFailedState.with(
+    {
+      groups: context.groups,
+      channels: context.channels,
+      cursor: context.cursor,
+      reason: event.payload,
+    },
+    [emitStatus({ category: categoryConstants.PNConnectionErrorCategory, error: event.payload?.message })],
+  ),
 );
 
 HandshakeReconnectingState.on(disconnect.type, (context) =>
   HandshakeStoppedState.with({
     channels: context.channels,
     groups: context.groups,
+    cursor: context.cursor,
+  }),
+);
+
+HandshakeReconnectingState.on(subscriptionChange.type, (_, event) =>
+  HandshakingState.with({ channels: event.payload.channels, groups: event.payload.groups }),
+);
```

Comment:
It looks like in this case we will throw away `cursor` which user potentially could pass in the other state, and we were keeping it for transition to the `Receiving` state.

```suggestion
HandshakeReconnectingState.on(subscriptionChange.type, (context, event) =>
  HandshakingState.with({ 
    channels: event.payload.channels, 
    groups: event.payload.groups,
    cursor: context.cursor,
  }),
);
```

Reply:
committed changes ✔️